tools/src/main/scala/org/apache/spark/sql/rapids/tool/qualification/QualAppInfo.scala

```diff
@@ -184,6 +185,30 @@ class QualAppInfo(
     }.getOrElse(1.0)
   }
 
+  def reportNestedType(): String = {
+    if (dataSourceInfo.size != 0) {
+      val schema = dataSourceInfo.map { ds => ds.schema }
+      val nestedSchema = parseReadSchemaForNestedTypes(schema.mkString(";"))
+      nestedSchema
+    } else {
+      ""
+    }
+  }
+
+  def parseReadSchemaForNestedTypes(schema: String): String = {
+    val distinctSchemas = schema.split(";").distinct
+    val schemaTypes = distinctSchemas.map(x => x.split(":", 2)(1))
+
+    // check for nested types in schema
+    val filterNestedTypes = schemaTypes.map { x =>
+      if (x.contains("array<") | x.contains("struct>") | x.contains("map")) x else ""
```

Comment:
is the > in struct> backwards?  I would have expected struct<?  If so we must be missing test for this

Reply:
I was expecting a much more complex parser here, looking for inner types as well.  This should just be traversing the schema maching up <> .  I would really like to find nested complex types as well. like array of array and array of map, etc..   I'm fine if we pull this into another pr.